addons/token-exchange/manager.go

```diff
@@ -0,0 +1,98 @@
+package addons
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	replicationv1alpha1 "github.com/csi-addons/volume-replication-operator/api/v1alpha1"
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	"golang.org/x/sync/errgroup"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	mgrScheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(mgrScheme))
+	utilruntime.Must(clusterv1.AddToScheme(mgrScheme))
+
+	utilruntime.Must(multiclusterv1alpha1.AddToScheme(mgrScheme))
+	utilruntime.Must(addonapiv1alpha1.AddToScheme(mgrScheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func getClient(config *rest.Config) (client.Client, error) {
+	scheme := runtime.NewScheme()
```

Comment:
Use mgrScheme instead of creating a new one. That means hub and spoke clients will understand the same APIs.